contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,666 @@
+    /// @dev Callable by the pool manager
+    /// @param _token The token to withdraw
+    function withdraw(address _token) external virtual onlyPoolManager(msg.sender) {
+        uint256 amount = _getBalance(_token, address(this));
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        _transferAmount(pool.token, msg.sender, amount);
+    }
+
+    /// @notice Sets the start and end dates.
+    /// @dev The timestamps are in seconds for the start and end times. The 'msg.sender' must be a pool manager.
+    ///      Emits a 'TimestampsUpdated()' event.
+    /// @param _registrationStartTime The start time for the registration
+    /// @param _registrationEndTime The end time for the registration
+    function updatePoolTimestamps(
+        uint64 _allocationStartTime,
+        uint64 _allocationEndTime,
+        uint64 _registrationStartTime,
+        uint64 _registrationEndTime
+    ) external onlyPoolManager(msg.sender) {
+        // If the timestamps are invalid this will revert - See details in '_isPoolTimestampValid'
+        _isPoolTimestampValid(_allocationStartTime, _allocationEndTime, _registrationStartTime, _registrationEndTime);
+
+        // Set the updated timestamps
+        registrationStartTime = _registrationStartTime;
+        registrationEndTime = _registrationEndTime;
+
+        // Emit that the timestamps have been updated with the updated values
+        emit TimestampsUpdated(
+            allocationStartTime, allocationEndTime, registrationStartTime, registrationEndTime, msg.sender
+        );
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Submit a proposal to LTIP pool
+    /// @dev Emits a 'Registered()' event
+    /// @param _data The data to be decoded
+    ///     - If registryGating is true, then the data is encoded as (address recipientId, address recipientAddress, uint256 allocationAmount, Metadata metadata)
+    ///     - If registryGating is false, then the data is encoded as (address recipientAddress, address registryAnchor, uitn256, allocationAmount, Metadata metadata)
+    /// @custom:data-registry (address recipientId, address recipientAddress, uint256 allocationAmount, Metadata metadata)
+    /// @custom:data-no-registry (address recipientAddress, address registryAnchor, uint256 allocationAmount, Metadata metadata)
+    /// @param _sender The sender of the transaction
+    /// @return recipientId The id of the recipient
+    function _registerRecipient(bytes memory _data, address _sender)
+        internal
+        override
+        onlyActiveRegistration
+        returns (address recipientId)
+    {
+        bool isUsingRegistryAnchor;
+        address recipientAddress;
+        address registryAnchor;
+        uint256 allocationAmount;
+        Metadata memory metadata;
+
+        // decode data custom to this strategy
+        if (registryGating) {
```

Comment:
^ just to add context, if we know the will always registry or never use registry, we could clean up the code (but if we are un-sure , this makes sense)

Reply:
For reference from the convo, this strategy should be generalizable so we'll leave this toggle here